Maybe I'm doing something wrong. How did you test this ?

```
ERROR:colcon:colcon metadata: unsupported operand type(s) for -=: 'set' and 'list'
Traceback (most recent call last):
  File "/home/mikael/work/ros2/colcon/install/colcon-core/lib/python3.5/site-packages/colcon_core/command.py", line 358, in verb_main
    rc = context.args.main(context=context)
  File "/home/mikael/work/ros2/colcon/install/colcon-metadata/lib/python3.5/site-packages/colcon_metadata/subverb/update.py", line 142, in main
    repository_name=name)
TypeError: unsupported operand type(s) for -=: 'set' and 'list'
```

--- a/colcon_metadata/subverb/remove.py
+++ b/colcon_metadata/subverb/remove.py
@@ -1,0 +1,37 @@
+# Copyright 2016-2018 Dirk Thomas
+# Licensed under the Apache License, Version 2.0
+
+from colcon_core.plugin_system import satisfies_version
+from colcon_metadata.metadata.repository import get_repositories
+from colcon_metadata.metadata.repository import set_repositories
+from colcon_metadata.subverb import MetadataSubverbExtensionPoint
+
+
+class RemoveMetadataSubverb(MetadataSubverbExtensionPoint):
+    """Remove a repository from the list of indexes."""
+
+    def __init__(self):  # noqa: D107
+        super().__init__()
+        satisfies_version(
+            MetadataSubverbExtensionPoint.EXTENSION_POINT_VERSION, '^1.0')
+
+    def add_arguments(self, *, parser):  # noqa: D102
+        argument = parser.add_argument(
+            'name',
+            help='The unique name identifying the repository')
+        try:
+            from argcomplete.completers import ChoicesCompleter
+        except ImportError:
+            pass
+        else:
+            repos = get_repositories()
+            argument.completer = \
+                ChoicesCompleter(repos.keys())
+
+    def main(self, *, context):  # noqa: D102
+        repos = get_repositories()
+        if context.args.name not in repos.keys():
+            return "A repository with the name '{context.args.name}' " \
+                "doesn't exist".format_map(locals())
+        del repos[context.args.name]
+        set_repositories(repos)

--- a/colcon_metadata/subverb/update.py
+++ b/colcon_metadata/subverb/update.py
@@ -5,6 +5,7 @@ import os
 import sys
 
 from colcon_core.plugin_system import satisfies_version
+from colcon_metadata.metadata import get_metadata_files
 from colcon_metadata.metadata import get_metadata_path
 from colcon_metadata.metadata.repository import get_repositories
 from colcon_metadata.metadata.repository import get_repository_metadata_files
@@ -127,11 +128,20 @@ class UpdateMetadataSubverb(MetadataSubverbExtensionPoint):
                 with destination_path.open('w') as h:
                     h.write(content)
 
-            # remove obsolete metadata files
+            # remove / rename obsolete metadata files
             for metadata_file in metadata_files_before:
                 if os.path.basename(metadata_file) not in metadata_basenames:
                     os.rename(metadata_file, metadata_file + '.obsolete')
                     print('  - {metadata_file} -> *.obsolete'
                           .format_map(locals()))
+
+        # remove / rename metadata files from obsolete repositories
+        obsolete_files = set(get_metadata_files())
+        for name in repos.keys():
+            obsolete_files -= set(get_repository_metadata_files(
+                repository_name=name))
+        for metadata_file in sorted(obsolete_files):
+            os.rename(metadata_file, metadata_file + '.obsolete')
+            print('  - {metadata_file} -> *.obsolete'.format_map(locals()))
 
         return rc

--- a/setup.cfg
+++ b/setup.cfg
@@ -58,4 +58,5 @@ colcon_core.verb =
 colcon_metadata.subverb =
     add = colcon_metadata.subverb.add:AddMetadataSubverb
     list = colcon_metadata.subverb.list:ListMetadataSubverb
+    remove = colcon_metadata.subverb.remove:RemoveMetadataSubverb
     update = colcon_metadata.subverb.update:UpdateMetadataSubverb

